### ISSUE TITLE
Line position of first line is wrong

### ISSUE THREAD

**Issue by tpecholt** [commonmark/cmark#274]
When I parse the following literal: `"*first*\nsecond"`

I get 
* CMARK_NODE_DOCUMENT 
  * CMARK_NODE_PARAGRAPH (start_line = 1, end_line = 2) 
    * CMARK_NODE_EMPH (**start_ line = 2, end_line = 2**)
    * CMARK_NODE_TEXT (start_line =2, end_line = 2)

the emph node should start and end at line = 1

**Comment by jgm** [comment]
Hm, with `cmark --sourcepos -t xml` I get
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">
<document sourcepos="1:1-2:6" xmlns="http://commonmark.org/xml/1.0">
  <paragraph sourcepos="1:1-2:6">
    <emph sourcepos="2:-7-2:-1">
      <text sourcepos="1:2-1:6" xml:space="preserve">first</text>
    </emph>
    <softbreak />
    <text sourcepos="2:1-2:6" xml:space="preserve">second</text>
  </paragraph>
</document>
```
The negative column positions are also, obviously, wrong!

**Comment by jgm** [comment]
@kivikakk added source positions for inline elements in #228 and may have ideas about what is going wrong here...

**Comment by kivikakk** [comment]
:eyes: I have a failing test case setup for this, now working on a fix.

**Comment by kivikakk** [comment]
Fix at #275.
